examples/policy_gradient/reinforce/agent.py

```diff
@@ -2,48 +2,258 @@
+        obs_slice = tf.slice(
+            all_obs,
+            [self.batch_index] + [0] * len(self.preprocessor.shape),
+            [self.per_device_batch_size] + [-1] * len(self.preprocessor.shape))
+        obs_slice.set_shape(observations.shape)
+        adv_slice = tf.slice(
+            all_adv, [self.batch_index], [self.per_device_batch_size])
+        acts_slice = tf.slice(
+            all_acts,
+            [self.batch_index] + [0] * len(self.action_shape),
+            [self.per_device_batch_size] + [-1] * len(self.action_shape))
+        plog_slice = tf.slice(
+            all_plog, [self.batch_index, 0], [self.per_device_batch_size, -1])
+        policy = ProximalPolicyLoss(
+            self.env.observation_space, self.env.action_space,
+            obs_slice, adv_slice, acts_slice, plog_slice, self.logit_dim,
+            self.kl_coeff, self.distribution_class, self.config, self.sess)
+        grads = self.optimizer.compute_gradients(
+            policy.loss, colocate_gradients_with_ops=True)
+
+      return Tower(
+          tf.group(
+              *[all_obs.initializer,
+                all_adv.initializer,
+                all_acts.initializer,
+                all_plog.initializer]),
+          grads,
+          policy)
+
+  def load_data(self, trajectories, full_trace):
+    """
+    Bulk loads the specified trajectories into device memory. The data is
+    split equally across all the devices.
+
+    Returns the number of tuples loaded per device.
+    """
+
+    truncated_obs = make_divisible_by(
+        trajectories["observations"], self.batch_size)
+    if full_trace:
+      run_options = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+    else:
+      run_options = tf.RunOptions(trace_level=tf.RunOptions.NO_TRACE)
+    run_metadata = tf.RunMetadata()
+    self.sess.run(
+        [t.init_op for t in self.towers],
+        feed_dict={
+            self.observations: truncated_obs,
+            self.advantages: make_divisible_by(
+                trajectories["advantages"], self.batch_size),
+            self.actions: make_divisible_by(
+                trajectories["actions"].squeeze(), self.batch_size),
+            self.prev_logits: make_divisible_by(
+                trajectories["logprobs"], self.batch_size),
+        },
+        options=run_options,
+        run_metadata=run_metadata)
+    if full_trace:
+      trace = timeline.Timeline(step_stats=run_metadata.step_stats)
+      trace_file = open('/tmp/ray/timeline-load.json', 'w')
+      trace_file.write(trace.generate_chrome_trace_format())
+
+    tuples_per_device = len(truncated_obs) / len(self.devices)
+    assert tuples_per_device % self.per_device_batch_size == 0
+    return tuples_per_device
+
+  def run_sgd_minibatch(self, batch_index, kl_coeff, full_trace, file_writer):
+    """
+    Runs a SGD step over the batch with index batch_index as created by
+    load_rollouts_data(), updating local weights.
+
```

Comment:
- first line of documentation should fit in one line (and come right after """)
- Colon after returns and then the next stuff should be on the next line